server/index.js

```diff
@@ -20,20 +20,38 @@ if (process.env.NODE_ENV === 'development') {
 
 app.use(express.static(publicPath));
 
-app.get('/api/post', (req, res, next) => {
+app.get('/api/main', (req, res, next) => {
   const sql = `
   select *
   from "post"
   `;
-  db
+  return db
     .query(sql)
     .then(result => {
-      return res.status(201).json(result.rows);
+      res.status(201).json(result.rows);
     })
     .catch(err => next(err));
 });
+
 app.use(express.json());
 
+app.get('/api/post/:postId', (req, res, next) => {
+  const targetId = Number(req.params.postId);
+  if (!Number.isInteger(targetId) || targetId <= 0) {
+    throw new ClientError(400, 'postId must be a positive integer!');
+  }
+  const sql = `
+  select*
+  from "post"
+  where "postId" = ${targetId}
+  `;
```

Comment:
In general, we do not put variables into the SQL template. Instead, use a parameterized query

https://node-postgres.com/features/queries#:~:text=stack))-,Parameterized%20query,-If%20you%20are